packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/NotificationData.kt

```diff
@@ -0,0 +1,64 @@
+package expo.modules.notifications.notifications.model
+
+import org.json.JSONArray
+import org.json.JSONObject
+
+/*
+* In some scenarios, data-only notifications are, in fact, presented.
+* */
+@JvmInline
+value class NotificationData(private val data: Map<String, String>) {
+  val title: String?
+    get() = data["title"]
+
+  val message: String?
+    get() = data["message"]
+
+  val body: JSONObject?
+    get() = try {
+      data["body"]?.let { JSONObject(it) }
+    } catch (e: Exception) {
+      null
+    }
+
+  val sound: String?
+    get() = data["sound"]
+
+  val shouldPlayDefaultSound: Boolean
+    get() = sound == null
+
+  val shouldUseDefaultVibrationPattern: Boolean
+    get() = data["vibrate"]?.toBoolean() == true
+
+  val isSticky: Boolean
+    get() = data["sticky"]?.toBoolean() ?: false
+
+  val vibrationPattern: LongArray?
+    get() = try {
+      data["vibrate"]?.let { vibrateString ->
+        JSONArray(vibrateString).let { jsonArray ->
+          LongArray(jsonArray.length()) { i ->
+            jsonArray.getLong(i)
+          }
+        }
+      }
+    } catch (e: Exception) {
+      // most likely a boolean value that cannot be converted to a longArray
+      null
+    }
+
+  val autoDismiss: Boolean
+    get() = data["autoDismiss"]?.toBoolean() ?: true
+
+  val categoryId: String?
+    get() = data["categoryId"]
+
+  val sticky: Boolean
+    get() = data["sticky"]?.toBoolean() ?: false
+
+  val subText: String?
+    get() = data["subtitle"]
+
+  val badge: Int?
+    get() = data["badge"]?.toIntOrNull()
+}
```

Comment:
The old `JSONNotificationContentBuilder` and [documentation](https://docs.expo.dev/versions/latest/sdk/notifications/#android-push-notification-payload-specification) also implement the following properties for `data`:

* priority
* color

For full backward compatibility I think they should also be used as fallback from `data`.

Reply:
Hello, thanks for flagging that. I added support for `color` but won't be adding support for `priority` in this PR because AFAICT, it should not be set in the `data` part of the notification but rather on the top-level object.

https://github.com/expo/expo-server-sdk-node/blob/7d620a833fd99a3071d69394b73b7182b6147397/src/ExpoClient.ts#L362